ha/monitor/k8s/parser.py

```diff
@@ -156,13 +153,49 @@ def _create_health_alert(self, res_type, res_name, health_status, generation_id)
         self.event.set_specific_info({"generation_id": generation_id})
         return self.event.json
 
-    def parse(self, an_event, cached_state):
+    @staticmethod
+    def _get_resource_id(labels: dict, resource_id_map: dict) -> str:
+        """
+        get resource id
+
+        Check if pod name label exist if yes then check if resource map contains it
+        if resource id doesn't found then check if machine id label is exists if yes
+        then find resource id for same.
+
+        Note: machine id in label is different from the actual machine id inside pod
+        in file '/etc/machine-id' and the actual machine id is resource id.
+
+        Args:
+            labels (dict): pod labels with values
+            resource_id_map (dict): map of label (name/machine-id) with actual machine id
+            label 'statefulset.kubernetes.io/pod-name' : pod fqdn, key in resource map
+            label 'cortx.io/machine-id' : uniq id, key in resource map
+        Returns:
+            str: resource id which is nothing but actual machine id
+                inside pod in file '/etc/machine-id'
+        """
+        resource_id = None
+        # NOTE: All the cortx pods from the deployment that are being watched
+        # will have either label 'cortx.io/machine-id' or 'statefulset.kubernetes.io/pod-name'
+        # both label will not co-exist in a deployment
+        # TODO: CORTX-31875 remove check for label 'cortx.io/machine-id'
+        if K8SEventsConst.LABEL_PODNAME in labels:
```

Comment:
if loop is redundant.

Reply:
done

Reply:
Hi,
Probably I don't understand what you are trying to say and, but here I don't think it's redundant, why:
1. there are 2 dict labels and resource_id_map
2. I need to check first if the labels contain the value for 'cortx.io/machine-id' if yes then take its value and again check if it exists in resource_id_map and if yes set that value to the resource_id.
i.e. if both dict has value then it will be looks like this **resource_id_map[labels[K8SEventsConst.LABEL_PODNAME]]**
3. same if added for machine_id (and here if pod label had key then machine_id will not exist in labels).

Reply:
@mssawant ,  the resource_id_map.get(labels[K8SEventsConst.LABEL_PODNAME], None)  in function _get_resource_id extracts and returns resource id either 
from LABEL_PODNAME (lie 178) OR 
from LABEL_MACHINEID ( line 186)  if its not found in podname dict. 

I see that the code has changed since comment was given
And it tries to get resourceid from machineid only if its not found in podname dict

Reply:
One thing here @mukhtar-inamdar , do we also need to return whether resource id was found using pod names or machine-ids ?

Reply:
@ArchanaLimaye No, it's just to print, to know if we correctly map it.
later in the next task CORTX-31875, we can remove the log also as we will be removing 1 block from here.